diesel/src/connection/sqlite/raw.rs

```diff
@@ -0,0 +1,70 @@
+extern crate libsqlite3_sys as ffi;
+extern crate libc;
+
+use std::ffi::{CString, CStr};
+use std::{ptr, str};
+
+use result::*;
+use result::Error::DatabaseError;
+
+pub struct RawConnection {
+    pub internal_connection: *mut ffi::sqlite3,
+}
+
+impl RawConnection {
+    pub fn establish(database_url: &str) -> ConnectionResult<Self> {
+        let mut conn_pointer = ptr::null_mut();
+        let database_url = try!(CString::new(database_url));
+        let connection_status = unsafe {
+            ffi::sqlite3_open(database_url.as_ptr(), &mut conn_pointer)
+        };
+
+        match connection_status {
+            ffi::SQLITE_OK => Ok(RawConnection {
+                internal_connection: conn_pointer,
+            }),
+            err_code => {
+                let message = super::error_message(err_code);
+                Err(ConnectionError::BadConnection(message.into()))
+            }
+        }
+    }
+
+    pub fn exec(&self, query: &str) -> QueryResult<()> {
+        let mut err_msg = ptr::null_mut();
+        let query = try!(CString::new(query));
+        unsafe {
+            ffi::sqlite3_exec(
+                self.internal_connection,
+                query.as_ptr(),
```

Comment:
These arguments to sqlite3_exec are pretty arcane. Is it easy to introduce either named variables or put comments here to explain what they do?